sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -158,6 +158,7 @@ object FileSourceStrategy extends Strategy with PredicateHelper with Logging {
         ExpressionSet(Nil)
       } else {
         val predicates = ExpressionSet(normalizedFilters
+          .flatMap(extractPredicatesWithinOutputSet(_, partitionSet))
           .filter(_.references.subsetOf(partitionSet)))
```

Comment:
After `extractPredicatesWithinOutputSet`, do we still need `.filter(_.references.subsetOf(partitionSet)))`?